prac_08/convert_miles_km.py

```diff
@@ -0,0 +1,41 @@
+"""
+Convert miles to km.
+"""
+
+from kivy.app import App
+from kivy.lang import Builder
+
+MILES_TO_KM = 1.60934
+
+
+class MilesConverterApp(App):
+    """Convert miles to km app"""
+
+    def build(self):
+        """build the Kivy app"""
+        self.title = "Convert Miles to Kilometres"
+        self.root = Builder.load_file('convert_miles_km.kv')
+        return self.root
+
+    def handle_calculate(self):
+        """calculate the conversion from mile to km"""
+        value = self.get_validated_miles()
+        result = value * MILES_TO_KM
+        self.root.ids.output_label.text = str(result)
+
+    def handle_increment(self, change):
+        """increase or decrease value by 1"""
+        value = self.get_validated_miles() + change
+        self.root.ids.input_miles.text = str(value)
+        self.handle_calculate()
+
+    def get_validated_miles(self):
+        """get value from text input and make sure it's a valid number"""
+        try:
+            value = float(self.root.ids.input_miles.text)
+            return value
```

Comment:
I don't think you have to have the return statement here, having value = function() should be enough to bring that value back